src/Sylius/Behat/Context/Ui/ProductContext.php

```diff
@@ -52,4 +82,26 @@ public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
 
         expect($this->productShowPage->isOpen(['product' => $product]))->toBe(false);
     }
+
+    /**
+     * @When I delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
```

Comment:
Shouldn't we set deleted product to ``sharedStorage`` at the end of method? If any error occurs in ``open`` of ``deleteProduct`` function and product will not be deleted, it should not be set in ``sharedStorage``, I suppose.